compiler/rustc_mir_transform/src/shim.rs

```diff
@@ -1379,33 +1379,19 @@ impl<'tcx> AsyncDestructorCtorShimBuilder<'tcx> {
     fn put_operand(&mut self, operand: Operand<'tcx>) {
         if let Some(top_cleanup_bb) = &mut self.top_cleanup_bb {
             let source_info = self.source_info;
-            self.bbs.raw.reserve(2);
             match &operand {
                 Operand::Copy(_) | Operand::Constant(_) => {
-                    for _ in 0..2 {
-                        *top_cleanup_bb = self.bbs.push(BasicBlockData {
-                            statements: Vec::new(),
-                            terminator: Some(Terminator {
-                                source_info,
-                                kind: TerminatorKind::Goto { target: *top_cleanup_bb },
-                            }),
-                            is_cleanup: true,
-                        });
-                    }
-                }
-                Operand::Move(place) => {
-                    let local = place.as_local().unwrap();
                     *top_cleanup_bb = self.bbs.push(BasicBlockData {
-                        statements: vec![Statement {
-                            source_info,
-                            kind: StatementKind::StorageDead(local),
-                        }],
```

Comment:
You could still do this, but in the same block as the `Drop` terminator.

No mir-opt tests seem to have changed though, so maybe add a mir-opt test, too?

Reply:
> You could still do this, but in the same block as the Drop terminator.

No, it's `Drop` (term) then `StorageDead` (stmt), while basic blocks execute statements first.